packages/fxa-auth-server/scripts/stripe-products-and-plans-to-firestore-documents/stripe-product-and-plans-converter.ts

```diff
@@ -0,0 +1,344 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+import { Logger } from 'mozlog';
+import Stripe from 'stripe';
+import { Container } from 'typedi';
+
+import {
+  BaseConfig,
+  CapabilityConfig,
+  StyleConfig,
+  StyleConfigKeys,
+  SupportConfig,
+  SupportConfigKeys,
+  UiContentConfig,
+  UiContentConfigKeys,
+  UrlConfig,
+  UrlConfigKeys,
+} from '../../lib/payments/configuration/base';
+import { PaymentConfigManager } from '../../lib/payments/configuration/manager';
+import { PlanConfig } from '../../lib/payments/configuration/plan';
+import { ProductConfig } from '../../lib/payments/configuration/product';
+import { commaSeparatedListToArray } from '../../lib/payments/configuration/utils';
+import { StripeHelper } from '../../lib/payments/stripe';
+
+/**
+ * TODO: write JSDOC
+ * Handles conversion and updates of Stripe metadata into Firestore documents.
+ * Documents could already exist and just need updating.
+ */
+export class StripeProductsAndPlansConverter {
+  private supportedLanguages: any;
+  private log: Logger;
+  private stripeHelper: StripeHelper;
+  private paymentConfigManager: PaymentConfigManager;
+
+  constructor({
+    log,
+    stripeHelper,
+    supportedLanguages,
+  }: {
+    log: Logger;
+    stripeHelper: StripeHelper;
+    supportedLanguages: any;
+  }) {
+    this.log = log;
+    this.stripeHelper = stripeHelper;
+    this.supportedLanguages = supportedLanguages.map((l: string) =>
+      l.toLowerCase()
+    );
+    this.paymentConfigManager = Container.get(PaymentConfigManager);
+  }
+
+  // Get all of the metadata keys that start with the prefix
+  getArrayOfStringsFromMetadataKeys(
+    metadata: Stripe.Product['metadata'],
+    metadataKeyPrefix: string
+  ): string[] {
+    // Get all of the metadata keys that start with the prefix
+    const metadataValues = [];
+    for (const [metadataKey, metadataValue] of Object.entries(metadata)) {
+      if (metadataKey.startsWith(metadataKeyPrefix)) {
+        metadataValues.push(metadataValue);
+      }
+    }
+    return metadataValues;
+  }
+
+  // TODO JSDoc
+  // For each metadata key that starts with "capabilities", add it to the returned object
+  // TODO Will we reuse these same methods for Plan metadata that overrides
+  // product metadata? E.g. I think MDN uses capabilities on plans for product tiers.
+  capabilitiesMetadataToCapabilityConfig(
+    product: Stripe.Product
+  ): CapabilityConfig {
+    const capabilities: CapabilityConfig = {};
+    for (const capability of Object.keys(product.metadata).filter((key) =>
+      key.toLowerCase().startsWith('capabilities')
+    )) {
+      capabilities[capability] = commaSeparatedListToArray(
+        product.metadata[capability]
+      );
+    }
+    return capabilities;
+  }
```

Comment:
This doesn't return the capabilities in the right shape I think.  With this the object is `{'capabilities': [capability_values], 'capabilities:CLIENT_ID_A': [capability_values], 'capabilities:CLIENT_ID_B': [capability_values], ...}` but I believe what we want is `{'*': [capability_values], 'CLIENT_ID_A': [capability_values], 'CLIENT_ID_B': [capability_values], ...}`

Reply:
You're right thank you. Fixed.